src/ray/gcs/tables.cc

```diff
@@ -468,29 +504,15 @@ Status ClientTable::Connect(const ClientTableDataT &local_client) {
     // Callback for a notification from the client table.
     auto notification_callback = [this](
         AsyncGcsClient *client, const UniqueID &log_key,
+        const GcsTableNotificationMode notification_mode,
         const std::vector<ClientTableDataT> &notifications) {
       RAY_CHECK(log_key == client_log_key_);
-      std::unordered_map<std::string, ClientTableDataT> connected_nodes;
-      std::unordered_map<std::string, ClientTableDataT> disconnected_nodes;
-      for (auto &notification : notifications) {
-        // This is temporary fix for Issue 4140 to avoid connect to dead nodes.
-        // TODO(yuhguo): remove this temporary fix after GCS entry is removable.
-        if (notification.is_insertion) {
-          connected_nodes.emplace(notification.client_id, notification);
-        } else {
-          auto iter = connected_nodes.find(notification.client_id);
-          if (iter != connected_nodes.end()) {
-            connected_nodes.erase(iter);
-          }
-          disconnected_nodes.emplace(notification.client_id, notification);
+      // We only handle the message of APPEND_OR_ADD and ignore the deletion message.
+      if (notification_mode == GcsTableNotificationMode::APPEND_OR_ADD) {
```

Comment:
The `notification_mode` should always be `GcsTableNotificationMode::APPEND_OR_ADD`, right? Can we make this a check instead of an `if`?

Reply:
We cannot do that, this is a callback for `RequestNotifications`. There will be messages for both `APPEND_OR_ADD` and `REMOVE`.